startup.py

```diff
@@ -139,16 +137,19 @@ def prepare_launch(self, exec_path, args, file_to_open=None):
         for option in options:
             file_dialog.setOption(option)
         # browse!
-        if file_dialog.exec_():
-            selected_file_path = file_dialog.selectedFiles()[0]
-            print("ue_project_dir = {}".format(selected_file_path))
-
-        args += "-projectPath " +  selected_file_path
-        self.logger.debug("Launch environment: %s", pprint.pformat(required_env))
-        self.logger.debug("Launch arguments: %s", args)
-         
-        
-        return LaunchInformation(exec_path, args, required_env)
+        while True and file_dialog.exec_():
+                selected_file_path = file_dialog.selectedFiles()[0]
+                if os.path.exists(selected_file_path+"\Assets"):
+                    args += "-projectPath " +  selected_file_path
+                    self.logger.debug("Launch environment: %s", pprint.pformat(required_env))                
+                    self.logger.debug("Launch arguments: %s", args)
+                    return LaunchInformation(exec_path, args, required_env)
+                else:
+                     error_message = QErrorMessage()
+                     error_message.showMessage("Invalid Unity Directory. Please Choose a Valid Unity Project")
+                     error_message.exec_()
```

Comment:
does this brings back the file choosing dialog?

Reply:
Thats what the while true is for. Once they close it, the file dialog relaunchs